pyomo/contrib/solver/solvers/highs.py

```diff
@@ -96,6 +96,121 @@ def update(self):
+
+    def _initialize_quad_coef_dict(self):
+        for coef in self.quadratic_coefs:
+            v1_ndx = coef.row_idx
+            v2_ndx = coef.col_idx
+            # Ensure we're storing the lower triangular part
+            row = max(v1_ndx, v2_ndx)
+            col = min(v1_ndx, v2_ndx)
+
+            coef_val = value(coef.expr)
+            # Adjust for diagonal elements
+            if v1_ndx == v2_ndx:
+                coef_val *= 2.0
+
+            self.quad_coef_dict[(row, col)] = coef_val
+
+    def update(self):
+        """
+        Update the quadratic objective expression.
+        """
+        needs_quadratic_update = self._first_update
+
+        self.constant.update()
+        for coef in self.linear_coefs:
+            coef.update()
+
+        for ndx, coef in enumerate(self.quadratic_coefs):
+            current_val = value(coef.expr)
+            if current_val != self.last_quadratic_coef_values[ndx]:
+                needs_quadratic_update = True
+
+                v1_ndx = coef.row_idx
+                v2_ndx = coef.col_idx
+                row = max(v1_ndx, v2_ndx)
+                col = min(v1_ndx, v2_ndx)
+
+                # Adjust the diagonal to match Highs' expected format
+                if v1_ndx == v2_ndx:
+                    current_val *= 2.0
+
+                self.quad_coef_dict[(row, col)] = current_val
+
+                self.last_quadratic_coef_values[ndx] = current_val
+
+        # If anything changed, rebuild and pass the Hessian
+        if needs_quadratic_update:
+            self._build_and_pass_hessian()
+            self._first_update = False
+
+    def _build_and_pass_hessian(self):
+        """Build and pass the Hessian to HiGHS in CSC format"""
+        if not self.quad_coef_dict:
+            return
+
+        dim = self.highs.getNumCol()
+
+        # Build CSC format for the lower triangular part
+        hessian_value = []
+        hessian_index = []
+        hessian_start = [0] * dim
+
+        sorted_entries = sorted(
+            self.quad_coef_dict.items(), key=lambda x: (x[0][1], x[0][0])
+        )
+
+        last_col = -1
+        for (row, col), val in sorted_entries:
+            while col > last_col:
+                last_col += 1
+                if last_col < dim:
+                    hessian_start[last_col] = len(hessian_value)
+
+            # Add the entry
+            hessian_index.append(row)
+            hessian_value.append(val)
+
+        while last_col < dim - 1:
+            last_col += 1
+            hessian_start[last_col] = len(hessian_value)
+
+        nnz = len(hessian_value)
+        status = self.highs.passHessian(
+            dim,
+            nnz,
+            highspy.HessianFormat.kTriangular,
+            np.array(hessian_start, dtype=np.int32),
+            np.array(hessian_index, dtype=np.int32),
+            np.array(hessian_value, dtype=np.double),
+        )
+
+        if status != highspy.HighsStatus.kOk:
+            logger.warning(
```

Comment:
I think this should be an exception. Thoughts?

Reply:
Thinking a bit more about, pretty much every single function in highspy return a HighsStatus (https://github.com/ERGO-Code/HiGHS/blob/364c83a51e44ba6c27def9c8fc1a49b1daf5ad5c/highs/Highs.h#L962), 
However, if I look at the current usage in this file, we don't capture the status when changeRowBounds, changeCoeff, changeColCost, etc... Just for the sake of consistency, there's an argument to remove this check

Reply:
Good point. I guess I don't feel strongly one way or the other.